pygsti/objectivefns/objectivefns.py

```diff
@@ -940,14 +941,36 @@ def add_omitted_freqs(self, printer=None, force=False):
         """
         if self.firsts is None or force:
             # FUTURE: add any tracked memory? self.resource_alloc.add_tracked_memory(...)
-            self.firsts = []; self.indicesOfCircuitsWithOmittedData = []
-            for i, c in enumerate(self.circuits):
-                indices = _slct.to_array(self.layout.indices_for_index(i))
-                lklen = _slct.length(self.layout.indices_for_index(i))
-                if 0 < lklen < self.model.compute_num_outcomes(c):
+            self.firsts = [] 
+            self.indicesOfCircuitsWithOmittedData = []
+
+            if self.outcome_count_by_circuit_cache is None:
```

Comment:
Same comment as in matrix layout: Can an empty `self.outcome_count_by_circuit_cache` simplify branching since you are using `get` to fall back to old behavior?

Reply:
In this particular instance the code path when the cache isn't present is meaningfully different than what is invoked as the fallback behavior when a value is missing from the cache (the bulk methods invoked in the former case are much faster).